src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -4124,9 +4511,9 @@ public override BoundNode VisitLocalFunctionStatement(BoundLocalFunctionStatemen
                     methodSignatureOpt: null,
                     returnTypes: null,
                     initialState: GetVariableState(),
-                    callbackOpt: _callbackOpt);
+                    analyzedNullabilityMapOpt: _disableNullabilityAnalysis ? null : _analyzedNullabilityMapOpt);
```

Comment:
>_disableNullabilityAnalysis ? null : _analyzedNullabilityMapOpt [](start = 47, length = 63)

this pattern appears in multiple places. What if we stored the map in a property, and the getter was checking `_disableNullabilityAnalysis` (if analysis is disabled, then you get no map...)?

Reply:
Could `_disableNullabilityAnalysis` be removed and replaced with `_analyzedNullabilityMapOpt == null`?

---
In reply to: [265755819](https://github.com/dotnet/roslyn/pull/34017#discussion_r265755819) [](ancestors = 265755819)

Reply:
`_disableNullabilityAnalysis` is intended to be removed: currently, it's used to suppress storing data from `BoundDeconstructionAssignment` and child nodes. Once those nodes are handled, it will no longer be needed, so I don't think it's worth trying to figure out a way to deduplicate this checking code now.

---
In reply to: [265757772](https://github.com/dotnet/roslyn/pull/34017#discussion_r265757772) [](ancestors = 265757772,265755819)